client/app/bundles/course/assessment/question/programming.jsx

```diff
@@ -0,0 +1,39 @@
+import React from 'react';
+import { render } from 'react-dom';
+import { Provider } from 'react-redux';
+import { IntlProvider, addLocaleData } from 'react-intl';
+
+import zh from 'react-intl/locale-data/zh';
+
+import createStore from './programming/store';
+import ProgrammingQuestion from './programming/containers/ProgrammingQuestion';
+import translations from '../../../../../build/locales/locales.json';
+
+const renderProgrammingQuestion = (props) => {
+  const i18nLocale = $("meta[name='server-context']").data('i18n-locale');
+  const availableForeignLocales = { zh };
+  const localeWithoutRegionCode = i18nLocale.toLowerCase().split(/[_-]+/)[0];
+
+  let messages;
+  if (localeWithoutRegionCode !== 'en' && availableForeignLocales[localeWithoutRegionCode]) {
+    addLocaleData(availableForeignLocales[localeWithoutRegionCode]);
+    messages = translations[localeWithoutRegionCode] || translations[i18nLocale];
+  }
+
+  const store = createStore(props);
+
+  render(
+    <Provider store={store}>
+      <IntlProvider locale={i18nLocale} messages={messages}>
+        <ProgrammingQuestion />
+      </IntlProvider>
+    </Provider>
+    , $('#programming-question')[0]
+  );
+};
+
+$.getJSON('', (data) => {
```

Comment:
How long does it take to build the data ?  If it's not very long it would be better to put the data in the initial request and parse them in the frontend, that will make the user side feels  faster.

LessonPlan pages uses 2 requests because the lesson plan data are quite large and it slows down the initial request a lot.